### ISSUE TITLE
HiTechnic NXT Color Sensor Version 2 support for EV3

### ISSUE THREAD

**Issue by bjost2s** [OpenRoberta/openroberta-lab#223]
User mentioned that they cannot use Nepo in competitions, because this sensor is not supported.
We have a couple of different systems for EV3 already, so the implementation has to be done for 

- [x] Blockly block
- [x] Serverside implementation
- [x] lejos v0
- [x] lejos v1
- [x] ev3dev
- [x] c4ev3

We could also consider to support it for NXT as well.

**Comment by simonedegiacomi** [comment]
@BeateJ @boonto  Specifically, which functions do we want to provide for this sensor?
These are the functions available through EV3-G:
![image](https://user-images.githubusercontent.com/15476739/63269290-2b5e6400-c296-11e9-958a-02d1c86cf816.png)

**Comment by boonto** [comment]
We tried it today, this should be the mode mapping:
- Color as colour mode
- W from RGB as light mode
- W from Passive as ambient light mode
- RGB as RGB mode
I just added the blockly block, however found we already had something implemented for NXT.
This wasnt that nice so I changed stuff slightly, but we have to discuss this again.
That is why Ill push my small changes to your c4ev3 branch so we dont commit half finished stuff.
You should be able to work on the rest.

**Comment by boonto** [comment]
Tested on c4ev3, ev3dev and lejos.

**Comment by boonto** [comment]
I forgot adding the wait untils, this leads to a server error.
Todo:
- [ ] Add wait until for HT sensor modes

**Comment by bjost2s** [comment]
There is still an issue with the codegeneration and checker, if there is the common colour sensor in the configuration connected instead of the HT Colour sensor:
here is an example with port 1 (common Colour Sensor) and port 2 (HT Colour Sensor) 
![image](https://user-images.githubusercontent.com/1613069/71764595-fface500-2ee9-11ea-9cba-0dbdf65ab536.png)

**Comment by boonto** [comment]
At the time of implementing this, we had decided to reuse the `ColorSensor` for this, to avoid creating a redundant `HTColorSensor`. However, this leads to problems like mentioned above with worse code with multiple `if (block is ht color)` clauses to catch every different behaviour.

I have now added the specific `HTColorSensor` which, although duplicating most of `ColorSensor` initially, leads to a much cleaner implementation.
